config/src/main/java/com/alibaba/nacos/config/server/service/ConfigDetailService.java

```diff
@@ -0,0 +1,187 @@
+ * @author 985492783@qq.com
+ * @date 2023/2/9 5:25
+ */
+@Service
+public class ConfigDetailService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConfigDetailService.class);
+    
+    private ConfigInfoPersistService configInfoPersistService;
+    
+    private BlockingQueue<SearchEvent> eventLinkedBlockingQueue = new LinkedBlockingQueue<>();
+    
+    private ScheduledExecutorService clientEventExecutor;
+    
+    private static final long WAIT_TIMEOUT = 4000L;
+    
+    private static final int MAX_THREAD = 2;
+    
+    public ConfigDetailService(ConfigInfoPersistService configInfoPersistService) {
+        this.configInfoPersistService = configInfoPersistService;
+        initWorker();
+    }
+    
+    private void initWorker() {
+        clientEventExecutor = new ScheduledThreadPoolExecutor(MAX_THREAD, r -> {
+            Thread t = new Thread(r);
+            t.setName("com.alibaba.nacos.config.search.worker");
+            t.setDaemon(true);
+            return t;
+        });
+        for (int i = 0; i < MAX_THREAD; i++) {
+            clientEventExecutor.submit(() -> {
+                while (true) {
+                    try {
+                        SearchEvent event = eventLinkedBlockingQueue.take();
+                        Page<ConfigInfo> result = null;
+                        if (Constants.CONFIG_SEARCH_BLUR.equals(event.getType())) {
+                            result = configInfoPersistService.findConfigInfoLike4Page(event.pageNo, event.pageSize,
+                                    event.dataId, event.group, event.tenant, event.configAdvanceInfo);
+                        } else {
+                            result = configInfoPersistService.findConfigInfo4Page(event.pageNo, event.pageSize,
+                                    event.dataId, event.group, event.tenant, event.configAdvanceInfo);
+                        }
+                        synchronized (event) {
+                            event.setResponse(result);
+                            event.notifyAll();
+                        }
+                    } catch (InterruptedException e) {
+                        LOGGER.error("catch search worker error: {}", e.getMessage());
+                    }
+                }
+            });
+        }
+    }
+    
+    /**
+     * block thread and use workerThread to search config.
+     */
+    public Page<ConfigInfo> findConfigInfoPage(String search, int pageNo, int pageSize, String dataId, String group,
+            String tenant, Map<String, Object> configAdvanceInfo) {
+        SearchEvent searchEvent = new SearchEvent(search, pageNo, pageSize, dataId, group, tenant,
+                configAdvanceInfo);
+        Page<ConfigInfo> result = null;
+        try {
+            synchronized (searchEvent) {
+                boolean offer = eventLinkedBlockingQueue.offer(searchEvent);
+                if (!offer) {
+                    throw new RuntimeException("config detail event offer fail.");
```

Comment:
NacosRuntimeException or NacosException replaced？
code maybe 503 means the server limit match.